de-lib/src/main/java/org/iplantc/de/client/models/sharing/SharedResource.java

```diff
@@ -8,11 +8,21 @@
     private String id;
 
     /**
+     * systemId of the shared resource
+     */
+    private String systemId;
+
+    /**
      * name of the shared resource
      */
     private String name;
 
     public SharedResource(String id, String name) {
+        this(null, id, name);
+    }
+
+    public SharedResource(String systemId, String id, String name) {
```

Comment:
In one of Dennis's previous commits to this PR, he introduced the model for `QualifiedId` which has an ID and system ID.  I'm not sure if it is or isn't worth putting here, just mentioning it.